Molly UI

Jenfer93 opened the pull request:
There are a few things that for sure needs to be looked at, 
But for now the UI is working, it's opening and closing as intended and sending messages. 

This to change later: 
* The custom CSS will need to be adjusted, and global styles implemented. 
* It forgets the messages when site is refreshed. 
* In chatMessage.svelte there needs to be a avatar for the user sending the message, and the name might be removed here. 
* It needs a way to sanitize the input so it's secure. 
<!-- This is an auto-generated comment: release notes by openai -->
### Summary by OpenAI

New Feature: A chat window UI has been added, including components for a logo, heading, input field, feed, and bubble. Messages are stored in an array and displayed in the feed. MollyChat, MollyAlert, MollyButton, MollyInput, MollyMessage, MollyMessages, MollyIcon, MollyWindow, and MollyWindowHead Svelte components have been added to support the chat UI. The changes are functional but need improvements such as global styles, message persistence, avatar display, and input sanitization.

> "A chat window appears,
> With bubbles and a logo dear.
> Molly helps you converse,
> With messages that won't disperse.
> New features to explore,
> And more improvements in store."
<!-- end of auto-generated comment: release notes by openai -->

pretzL commented on packages/molly/src/routes/chat/components/chatBubble.svelte:
```diff
@@ -0,0 +1,24 @@
+}
+</script>
+
+<style>
+  .chat-bubble {
+  position: fixed;
+  bottom: 0;
+  right: 0;
+  padding: 10px;
+  border-radius: 50px;
+  background-color: #323E47;
+  color: white;
+  cursor: pointer;
+  }
+</style>
```
Could some of these styles be added to the global scope?

Jenfer93 replied:
I'm not able to access the global styles, not sure why

pretzL commented on packages/molly/src/routes/chat/components/chatButton.svelte:
```diff
@@ -0,0 +1,31 @@
+
+</script>
+
+<style>
+  button {
+  background-color: #21A299;
+  border-radius: 10px;
+  width: 50px;
+  border: none;
+  color: white; 
+  cursor: pointer;
+  }
```
Love me some global styles

Fermain replied:
Svelte styles are scoped locally by default

Jenfer93 replied:
sure would love some globals yeah

pretzL commented on packages/molly/src/routes/chat/components/chatFeed.svelte:
```diff
@@ -0,0 +1,64 @@
+
+	onMount(() => {
+		//Scroll to the bottom of the chat window
+		const chatWindow: HTMLElement | null = document.querySelector('.chat-content');
```
OpenAI also complained about this, but using Svelte's own `this` selector is a really nice feature. Eliminates the need to querySelect anything (we also spoke about this in DMs, but I thought I'd add it as a comment as well just in case)

Jenfer93 replied:
I've pushed a change to this

pretzL commented on packages/molly/src/routes/chat/components/chatFeed.svelte:
```diff
@@ -0,0 +1,64 @@
+
+	//Watch for changes in the messages array and scroll to the bottom of the chat window
+	$: {
+		const chatWindow: HTMLElement | null = document.querySelector('.chat-content');
+		if (chatWindow) {
```
:wicked:

pretzL commented on packages/molly/src/routes/chat/logo/angle-small-down.svg:
I've added the ability to use Material Icons directly in the project. You can use them by creating an `i` element like so: `<i class="icon-f">arrow_down</i>`

Jenfer93 replied:
I don't know if I can access it, we will have to talk some more about that

pretzL commented on packages/molly/src/routes/chat/logo/paper-plane.svg:
I think Material Icons has something similar? If not you can continue to use this

pretzL reviewed (changes requested):
Check the comments. There are just a couple of tweaks we can make to ensure everyone's on the same page. Other than that, the chat window looks great!

Fermain commented on packages/molly/src/routes/+page.svelte:
```diff
@@ -1,3 +1,8 @@
+</script>
 <h1>Welcome to your library project</h1>
 <p>Create your package using @sveltejs/package and preview/showcase your work with SvelteKit</p>
 <p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>
```
```suggestion
```

Fermain commented on packages/molly/src/routes/+page.svelte:
```diff
@@ -1,3 +1,8 @@
+
+</script>
 <h1>Welcome to your library project</h1>
 <p>Create your package using @sveltejs/package and preview/showcase your work with SvelteKit</p>
```
```suggestion
```

Fermain commented on packages/molly/src/routes/+page.svelte:
```diff
@@ -1,3 +1,8 @@
+	import MollyChat from "./chat/mollyChat.svelte";
+
+</script>
 <h1>Welcome to your library project</h1>
```
```suggestion
```

Fermain commented on packages/molly/src/routes/chat/components/chatFeed.svelte:
```diff
@@ -0,0 +1,64 @@
+	}
+</script>
+
+<div class="chat-content">
```
```suggestion
let container: HTMLElement;
</script>

<div bind:this={container} class="chat-content">
```

Fermain commented on packages/molly/src/routes/chat/components/chatInput.svelte:
```diff
@@ -0,0 +1,95 @@
+
+	const dispatch = createEventDispatcher();
+
+	interface Message {
```
We could move this to `@mollify/types` as MollifyMessage, or use the types from the `OpenAI` package

Jenfer93 replied:
Either sounds good to me, not sure how you would like to do it!

Fermain reviewed (changes requested):
Some changes from me too, overlapping with Eric's comments

Jenfer93 commented:
STILL WORK IN PROGRESS! Just need a little break!

Fermain commented on packages/molly/src/lib/components/Molly.svelte:
```diff
@@ -0,0 +1,95 @@
+	let scrollToDiv: HTMLDivElement;
+	export let endpoint = '/';
+
+	function scrollToBottom() {
+		setTimeout(function () {
+			scrollToDiv.scrollIntoView({ behavior: 'smooth', block: 'end', inline: 'nearest' });
+		}, 100);
+	}
```
@openai I am seriously considering removing you.

jaaneh replied:
> @openai I am seriously considering removing you.

Do it. You won't.

Fermain replied:
#44 I am the one that knocks

Fermain commented on packages/molly/src/lib/components/Molly.svelte:
```diff
@@ -0,0 +1,95 @@
+<script lang="ts">
+	import MollyMessage from '$lib/components/MollyMessage.svelte';
+	import type { ChatCompletionRequestMessage } from 'openai';
+	import { SSE } from 'sse.js';
```
Yes, actually it is used in the code. Removing it would completely break everything. Please be more careful.

Jenfer93 commented:
I've changed some things, I deleted the unused components, files and folders, and I managed to move form out of the main component. It still won't let me merge, as you @Fermain and @pretzL requested changes, so would you like to look at it again, and tell me if I can merge or not :)

Fermain reviewed (approved):
Happy with this as MVP

Fermain commented:
@pretzL anything else you want to add here?

pretzL reviewed (approved):
Good stuff
